runtime/src/iree/vm/buffer.h

```diff
@@ -199,6 +199,12 @@ IREE_API_EXPORT iree_status_t iree_vm_buffer_write_elements(
     iree_host_size_t target_offset, iree_host_size_t element_count,
     iree_host_size_t element_length);
 
+// Computes the SipHash-2-4 of the source buffer along the byte range starting
+// as |source_offset| of |length| bytes. Uses 128-bit key `0x00010203...0e0f`.
+IREE_API_EXPORT iree_status_t iree_vm_buffer_hash(
+    const iree_vm_buffer_t* source_buffer, iree_host_size_t source_offset,
+    iree_host_size_t length, int64_t* result);
```

Comment:
if this is output only then prefix with `out_` - if it's I/O then just `result` is fine but we'd want to document the expectations (see comment about incremental hashing).

```suggestion
    iree_host_size_t length, int64_t* out_result);
```